HKMP/Game/Client/Entity/WanderingHusk.cs

```diff
@@ -0,0 +1,145 @@
+                    // Also include the clip name as event info
+                    firstFrame.eventInfo = clip.name;
+                }
+            }
+            else {
+                Logger.Get().Warn(this, "Walker animator not found");
+            }
+            // Making each animation send an update
+            walkerAnimator.AnimationEventTriggered = (caller, currentClip, currentFrame) => {
+                if (currentClip.name == "Idle") {
+                    SendAnimationUpdate((byte)Animation.Idle);
+                }
+                else if (currentClip.name == "Walk") {
+                    SendAnimationUpdate((byte)Animation.Walk);
+                }
+                else if (currentClip.name == "Turn") {
+                    SendAnimationUpdate((byte)Animation.Turn);
+                }
+            };
+        }
+
+        private void CreateAnimationEvents() {
+            _fsm.InsertMethod("Anticipate", 0, CreateUpdateMethod(() => { SendAnimationUpdate((byte)Animation.Anticipate); }));
+            _fsm.InsertMethod("Lunge", 0, CreateUpdateMethod(() => { SendAnimationUpdate((byte)Animation.Lunge); }));
+            _fsm.InsertMethod("Cooldown", 0, CreateUpdateMethod(() => { SendAnimationUpdate((byte)Animation.Cooldown); }));
+            _fsm.InsertMethod("Idle", 0, CreateUpdateMethod(() => { SendAnimationUpdate((byte)Animation.Idle); }));
+        }
+        protected override void InternalInitializeAsSceneHost() {
+            SendStateUpdate((byte)State.Active);
+            _walker.enabled = true;
+        }
+        protected override void InternalInitializeAsSceneClient(byte? stateIndex) {
+            RemoveAllTransitions(_fsm);
+            _walker.Stop(Walker.StopReasons.Bored); // Stops enemy from 'drifting' when entering the room. 
+            _walker.enabled = false;
+        }
+        protected override void InternalSwitchToSceneHost() {
+            RestoreAllTransitions(_fsm);
+            _walker.enabled = true;
+            // Put the FSM back into the correct state based on the last state. 
+            switch (_lastAnimation) {
+                case Animation.Walk:
+                    _fsm.SetState("Idle");
+                    break;
+                case Animation.Anticipate:
+                    _fsm.SetState("Lunge");
+                    break;
+                case Animation.Cooldown:
+                    _fsm.SetState("Idle");
+                    break;
+                case Animation.Lunge:
+                    _fsm.SetState("Cooldown");
+                    break;
+            }
+
+        }
+
+        public override void UpdateAnimation(byte animationIndex, byte[] animationInfo) {
+            base.UpdateAnimation(animationIndex, animationInfo);
+
+            var animation = (Animation)animationIndex;
+
+            _lastAnimation = animation;
+
+            if (animation == Animation.Anticipate) {
+                _fsm.ExecuteActions("Anticipate", 2, 3, 5);
```

Comment:
A few of the actions that you try to execute in this call are not implement in `ActionExtensions` in the `HKMP.Fsm` namespace. Therefore, they will not be executed and give a warning in the ModLog. I have implemented these actions in the `entity-system` branch, so after merging it they should work. This should also resolve the lunging sound not working client side.

Reply:
As a heads up for next time: every time you see such an error in the ModLog, feel free to implement the action in `ActionExtensions` similarly to the others that are already present.

Reply:
Thanks for the heads up! I'll be sure to pay more attention for my next entity